Documentation/ApiOverview/Country/Index.rst

```diff
@@ -0,0 +1,168 @@
+..  include:: /Includes.rst.txt
+..  index:: Country API
+..  _country-api:
+
+===========
+Country API
+===========
+
+..  versionadded:: 12.2
+
+TYPO3 ships a list of countries of the world. The list is based on the
+`ISO 3166-1`_ standard, with the alphanumeric short name ("FR" or "FRA" in its
+three-letter short name), the English name ("France"), the official name
+("Republic of France"), also the numerical code, and the country's flag as
+emoji (UTF-8 representation).
+
+..  note::
+    The country list is based on `Debian's ISO code list`_ and shipped
+    statically as PHP content in the Country API.
+
+..  _ISO 3166-1: https://en.wikipedia.org/wiki/ISO_3166-1
+..  _Debian's ISO code list: https://salsa.debian.org/iso-codes-team/iso-codes
+
+.. contents:: Contents
+   :local:
+
+
+Using the PHP API
+=================
+
+:ref:`Dependency injection <DependencyInjection>` can be used to retrieve the
+:php:`\TYPO3\CMS\Core\Country\CountryProvider` class:
+
+..  literalinclude:: _MyClass.php
+    :caption: EXT:my_extension/Classes/MyClass.php
+
+
+Get a country
+-------------
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/MyClass.php
+
+    // Get the country by Alpha-2 code
+    $france = $this->countryProvider->getByIsoCode('FR');
+
+    // Get the country by name
+    $france = $this->$countryProvider->getByName('France');
+
+    // Get the country by Alpha-3 code
+    $france = $this->$countryProvider->getByAlpha3IsoCode('FRA');
+
+The methods return a :php:`\TYPO3\CMS\Core\Country\Country` object.
+
+
+Get all countries
+-----------------
+
+To get all countries call the :php:`getAll()` method:
+
+..  code-block:: php
+    :caption: EXT:my_extension/Classes/MyClass.php
+
+    $allCountries = $this->countryProvider->getAll();
+
+The method returns an array of :php:`\TYPO3\CMS\Core\Country\Country` objects.
+
+
+The :php:`Country` object
+-------------------------
+
+A country object can be used to fetch all information about it, also with
+translatable labels:
+
+..  literalinclude:: _MyClassWithTranslation.php
+    :caption: EXT:my_extension/Classes/MyClassWithTranslation.php
+
+
+PHP API reference
+=================
+
+:php:`CountryProvider`
+----------------------
+
+.. include:: /CodeSnippets/Manual/Country/CountryProvider.rst.txt
+
+:php:`Country`
+--------------
+
+.. include:: /CodeSnippets/Manual/Country/Country.rst.txt
+
+
+..  index:: CountrySelect ViewHelper
+..  _country-select-viewhelper:
+
+Form ViewHelper
+===============
+
+A Fluid ViewHelper is shipped with TYPO3 to render a dropdown for forms:
+
+..  code-block:: html
+
+    <f:form.countrySelect
+        name="country"
+        value="AT"
+        sortByOptionLabel="true"
+        prioritizedCountries="{0: 'DE', 1: 'AT', 2: 'CH'}"
+    />
+
+Available options
```

Comment:
I think all these options belong in the viewhelper reference and we should link there

Reply:
Then we need a Core patch so that all available options are described in the docblock of the view helper. Or we say, one should just lookup up the arguments in the view helper class (like in the other view helpers).

BTW: Do you know why the new CountrySelectViewHelper is not listed here: https://docs.typo3.org/other/typo3/view-helper-reference/main/en-us/typo3/fluid/latest/Index.html

Reply:
We must check if the rst was generated and if so, if the rendering failed for some reason.

Did you compare weather the descriptions here give additional value to the names of the arguments in the viewhelper?

Reply:
I am not in those rendering parts and have no clue what should be done and where to check.

The current docblock for the view helper:
https://github.com/TYPO3/typo3/blob/main/typo3/sysext/fluid/Classes/ViewHelpers/Form/CountrySelectViewHelper.php#L25-L45

The arguments don't give the additional examples. So either we provide them here or we adjust the view helper.

Reply:
I am with you, that the view helper should hold the reference. But before it is not rendered, a link is not possible. Therefore, I set this PR to draft.